src/Tools/_framework/Paths/Courses.jsx

```diff
@@ -0,0 +1,566 @@
+// import axios from 'axios';
+import {
+  Box,
+  Button,
+  Text,
+  Grid,
+  GridItem,
+  Wrap,
+  Flex,
+  DrawerFooter,
+  FormErrorMessage,
+  Input,
+  FormLabel,
+  FormControl,
+  FormHelperText,
+  Stack,
+  DrawerBody,
+  DrawerHeader,
+  DrawerCloseButton,
+  DrawerContent,
+  DrawerOverlay,
+  Drawer,
+  useDisclosure,
+  Image,
+  Popover,
+  PopoverTrigger,
+  PopoverContent,
+  PopoverArrow,
+  PopoverBody,
+  SimpleGrid,
+  AlertDialog,
+  AlertDialogOverlay,
+  AlertDialogContent,
+  AlertDialogHeader,
+  AlertDialogBody,
+  AlertDialogFooter,
+} from "@chakra-ui/react";
+
+import React, { useEffect, useRef, useState } from "react";
+import { useLoaderData, useFetcher } from "react-router-dom";
+import axios from "axios";
+import { CourseCard } from "../../../_reactComponents/PanelHeaderComponents/CourseCard";
+import { driveColors, driveImages } from "../../../_reactComponents/Drive/util";
+
+export async function action({ request }) {
+  const formData = await request.formData();
+  let formObj = Object.fromEntries(formData);
+
+  if (formObj?._action == "Create New Course") {
+    let { data } = await axios.get("/api/createCourse.php");
+    if (!data.success) throw Error(data.message);
+    return true;
+  } else if (formObj?._action == "Delete") {
+    let { data } = await axios.post("/api/deleteCourse.php", {
+      courseId: formObj.courseId,
+    });
+    if (!data.success) throw Error(data.message);
+    return true;
+  } else if (formObj?._action == "Duplicate") {
+    let { data } = await axios.post("/api/duplicateCourse.php", {
+      courseId: formObj.courseId,
+      dateDifference: formObj.dateDifference,
+      newLabel: formObj.newLabel,
+    });
+    if (!data.success) throw Error(data.message);
+    return true;
+  } else if (formObj?._action == "Rename") {
+    // let { data } = await axios.post("/api/modifyCourse.php", {
+    let resp = await axios.post("/api/modifyCourse.php", {
+      courseId: formObj.courseId,
+      label: formObj.newLabel,
+    });
+    //TODO: modifyCourse.php doesn't respond with data.success
+    // console.log(resp);
+    // if (!data.success) throw Error(data.message);
+    return true;
+  } else if (formObj?._action == "Update Image") {
+    // let { data } = await axios.post("/api/modifyCourse.php", {
+
+    let resp = await axios.post("/api/modifyCourse.php", {
```

Comment:
It doesn't look like this code is handling the potential for non-success response codes, which can be returned from this API before or after the try/catch refactoring.

Reply:
A potential quick and dirty fix is just to use a browser alert on failure - https://github.com/Doenet/DoenetTools/blob/37de7bab0a5388b4390e88acb1b6ecf3b4ff0fcc/src/Tools/_framework/Paths/Community.jsx#L42

Reply:
Done